heareval/predictions/task_predictions.py

```diff
@@ -749,30 +779,153 @@ def dataloader_from_split_name(
 class GridPointResult:
     def __init__(
         self,
-        model_path: str,
         epoch: int,
-        time_in_min: float,
         hparams: Dict[str, Any],
         postprocessing: Tuple[Tuple[str, Any], ...],
-        trainer: pl.Trainer,
         validation_score: float,
         score_mode: str,
+        conf: Dict,
     ):
-        self.model_path = model_path
         self.epoch = epoch
-        self.time_in_min = time_in_min
         self.hparams = hparams
         self.postprocessing = postprocessing
-        self.trainer = trainer
         self.validation_score = validation_score
         self.score_mode = score_mode
+        self.conf = conf
 
 
-def task_predictions_train(
+def task_predictions_train_test_full(
+    best_grid_point: GridPointResult,
     embedding_path: Path,
     embedding_size: int,
-    grid_points: int,
     metadata: Dict[str, Any],
+    data_splits: Dict[str, List[str]],
+    label_to_idx: Dict[str, int],
+    nlabels: int,
+    scores: List[ScoreFunction],
+    gpus: Any,
+    in_memory: bool,
+    deterministic: bool,
+):
+    """
+    Trains on the full `train+valid` split to utilise all the available data
+    and tests on the `test` split.
+
+    Model configurations are copied from the best grid point which was found using the
+    `train` and the `valid` split.
+    """
+    start = time.time()
+
+    # Get the configuration of the best_grid_point and use it to train on the
+    # full train+valid set and test on the test set
+    conf = best_grid_point.conf
+    # Set the max_epochs to the number of epochs done in the best grid point
+    conf["max_epochs"] = best_grid_point.epoch
```

Comment:
I see. Instead of using early stopping we cap the number of epochs to the number of epochs reached during the grid point search. Are we sure that's what we want? This is definitely quicker. But, I'm not positive this is what we want to do.

Reply:
@turian

Reply:
@jorshi @khumairraj No this is not correct.

You generally fix max_epochs when you are retraining and combining train+val so that you have a bigger train and no val any more.

If we are doing k-fold and fixing the hyperparameters, but we have separate train and validation folds, we should use our validation fold for early stopping again.